src/main/java/io/supertokens/dashboard/Dashboard.java

```diff
@@ -0,0 +1,190 @@
+        DashboardSQLStorage storage = StorageLayer.getDashboardStorage(main);
+        try {
+            storage.startTransaction(transaction -> {
+                if (newEmail != null) {
+                    try {
+                        storage.updateDashboardUsersEmailWithEmail_Transaction(transaction, email, newEmail);
+                    } catch (DuplicateEmailException e) {
+                        throw new StorageTransactionLogicException(e);
+                    }
+                }
+
+                if (newPassword != null) {
+                    storage.updateDashboardUsersPasswordWithEmail_Transaction(transaction, email, newPassword);
+                }
+                storage.commitTransaction(transaction);
+                return null;
+            });
+        } catch (StorageTransactionLogicException e) {
+            if (e.actualException instanceof DuplicateEmailException) {
+                throw (DuplicateEmailException) e.actualException;
+            }
+            throw e;
+        }
+    }
+
+    public static void updateUsersCredentialsWithUserId(Main main, String userId, String newEmail, String newPassword)
+            throws StorageQueryException, DuplicateEmailException, StorageTransactionLogicException {
+        DashboardSQLStorage storage = StorageLayer.getDashboardStorage(main);
+        try {
+            storage.startTransaction(transaction -> {
+                if (newEmail != null) {
+                    try {
+                        storage.updateDashboardUsersEmailWithUserId_Transaction(transaction, userId, newEmail);
+                    } catch (DuplicateEmailException e) {
+                        throw new StorageTransactionLogicException(e);
+                    }
+                }
+
+                if (newPassword != null) {
+                    String hashedPassword = PasswordHashing.getInstance(main).createHashWithSalt(newPassword);
+                    storage.updateDashboardUsersPasswordWithUserId_Transaction(transaction, userId, hashedPassword);
+                }
+                storage.commitTransaction(transaction);
+                return null;
+            });
+        } catch (StorageTransactionLogicException e) {
+            if (e.actualException instanceof DuplicateEmailException) {
+                throw (DuplicateEmailException) e.actualException;
+            }
+            throw e;
+        }
+    }
+
+    public static boolean isFeatureFlagEnabledOrUserCountIsUnderThreshold(Main main) throws StorageQueryException {
+
+        if (!isDashboardFeatureFlagEnabled(main)) {
+            // retrieve current dashboard users
+            DashboardUser[] users = StorageLayer.getDashboardStorage(main).getAllDashboardUsers();
+            // check if current dashboard users count is under the threshold
+            return users.length <= MAX_NUMBER_OF_FREE_DASHBOARD_USERS;
+        }
+        return true;
+    }
```

Comment:
I prefer if it was false by default and then returns true if conditions are met. Less chance of making mistakes if this condition becomes more complex in the future